pandas/_typing.py

```diff
@@ -68,8 +68,13 @@
     "ExtensionDtype", str, np.dtype, Type[Union[str, float, int, complex, bool, object]]
 ]
 DtypeObj = Union[np.dtype, "ExtensionDtype"]
-FilePathOrBuffer = Union[str, Path, IO[AnyStr], IOBase]
-FileOrBuffer = Union[str, IO[AnyStr], IOBase]
+FilePathOrBuffer = Union[
+    str, Path, IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper
+]
+FileOrBuffer = Union[
+    str, IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper
+]
+Buffer = Union[IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper]
```

Comment:
isn't FileOrBuffer=Union[str, Buffer] ? (and similar for FilePathOrBuffer), might be easier to grok

Reply:
yes, that is the case but they take generics (`FileorBuffer[str]`) for `IO`. Mypy doesn't like nesting them - I'm not a typing expert, maybe there is a way to overcome that.